builtin/providers/aws/resource_aws_ssm_document.go

```diff
@@ -381,6 +408,47 @@ func deleteDocumentPermissions(d *schema.ResourceData, meta interface{}) error {
 	return nil
 }
 
+func updateAwsSSMDocument(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] Updating SSM Document: %s", d.Id())
+
+	name := d.Get("name").(string)
+
+	updateDocInput := &ssm.UpdateDocumentInput{
+		Name:            aws.String(name),
+		Content:         aws.String(d.Get("content").(string)),
+		DocumentVersion: aws.String(d.Get("default_version").(string)),
+	}
+
+	newDefaultVersion := d.Get("default_version").(string)
+
+	ssmconn := meta.(*AWSClient).ssmconn
+	updated, err := ssmconn.UpdateDocument(updateDocInput)
+
+	if isAWSErr(err, "DuplicateDocumentContent", "") {
+		log.Printf("[DEBUG] Content is a duplicate of the latest version so update is not necessary: %s", d.Id())
+		log.Printf("[INFO] Updating the default version to the latest version %s: %s", newDefaultVersion, d.Id())
+
+		newDefaultVersion = d.Get("latest_version").(string)
+	} else if err != nil {
+		return errwrap.Wrapf("Error updating SSM document: {{err}}", err)
+	} else {
+		log.Printf("[INFO] Updating the default version to the new version %s: %s", newDefaultVersion, d.Id())
+		newDefaultVersion = *updated.DocumentDescription.DocumentVersion
```

Comment:
Good point. I think it'll be fine in this case but I'll keep an eye out in future